### PR TITLE
[Conflicts][WIP] Fire! [requested contribution]

### PR DESCRIPTION
**Pull request by FilippoLeon**
Ever desired to burn down your spaceship? From now you can use fire!

fire

What is there:
 - fire spreads and destroys
 - integrated with the temperature system
 - need a "Post Update hook" to let LUA delete/place furniture
 - fire consumes oxygen and produces CO2
 - fire dies if pressure too low
 - randomly starts when mostly inconvenient
 - conveniently buildable by your spacers

What is not there (and you can help and fix):
 - the "fire spread" sound is the same as placing furniture
 - there is no "fire crackling" sound
 - some error is thrown if you try to spread to invalid position
 - sprite is horrible
 - who knows?

### CODE REVIEW

**Comment by sboigelot** [comment]
You should add in the todo list: fire will die in absence of oxygen.

File: Assets/StreamingAssets/Data/Furniture.xml [hunk 1]
```diff
@@ -373,6 +373,33 @@
         <UnlocalizedDescription>furn_oxygen_compressor_desc</UnlocalizedDescription>
 
     </Furniture>
+  
+  
+    <Furniture objectType="fire" skipBuildMenu="true">
+        <Name>Fire</Name>
+        <Description>Stores and releases oxygen as needed</Description>
```
Comment (sboigelot) [inline Assets/StreamingAssets/Data/Furniture.xml]: You may want to update the description ;)

Comment (FilippoLeon) [inline Assets/StreamingAssets/Data/Furniture.xml]: :P

**Comment by FilippoLeon** [comment]
Yes, good idea.

**Comment by Tritaris** [comment]
There is the event system that can be used to randomly have a fire start. It was added in PR #270. This would be a good way to have the fires start. I would have it when a machine malfunctions it has a chance to burst into flames. You can get a list of the different machines installing into the world from `World.current.furnitures`.

File: Assets/StreamingAssets/Data/Furniture.xml (hunk 1 above)
Comment (Mizipzor) [inline Assets/StreamingAssets/Data/Furniture.xml]: Yes, this needs fixing before merge.

**Comment by FilippoLeon** [comment]
Updated!

**Comment by bjubes** [comment]
not necessary for this PR, but flammability as a property on furniture and tiles would be a very cool addition

**Comment by sboigelot** [comment]
@FilippoLeon What your status on this? Do you still work on it or should someone take it over?

**Comment by sboigelot** [comment]
This was a great start, but the OP didn't give sign of life since 21 days. Closing this PR for now. @FilippoLeon feel free to open it again if you come back!

**Pull request closed without merging**